File: SecondYear/aayushraghav93_AayushRaghav_2226CSE(AI)1108_2/Week_4/firstmissingpositive.java
import java.util.* ;
import java.io.*; 
public class Solution {
	public static int firstMissing(int[] arr, int n) {
		// Write your code here.
		HashSet<Integer> positiveSet = new HashSet<>();
        for (int num : arr) {
            if (num > 0) {
                positiveSet.add(num);
            }
		}
        int missing = 1;
        while (positiveSet.contains(missing)) {
            missing++;
        }
        return missing;		
	}
}

File: SecondYear/any-mesh_AnimeshKumarSingh_2226CSE(AI&ML)1119_2/Week_4/FirstMissingPositive.cpp
#include <bits/stdc++.h> 
int firstMissing(int arr[], int n)
{
    // Write your code here.
    vector<int> ans;
    for(int i = 0 ; i < n ; i++){
        if(arr[i] > 0)
            ans.push_back(arr[i]);
    }
    sort(ans.begin(), ans.end());
    int min = INT_MAX;
    for(int i = 0 ; i < ans.size() ; i++){
        if(ans[i] != i + 1)
            return i + 1;
    }
    return ans.size() + 1;
}
